ts/pages/governance/modal_vote.tsx

```diff
@@ -1,238 +1,192 @@
+import { useAPIClient } from 'ts/hooks/use_api_client';
 import { ErrorModal } from 'ts/pages/governance/error_modal';
 import { VoteForm, VoteInfo } from 'ts/pages/governance/vote_form';
+import { Dispatcher } from 'ts/redux/dispatcher';
+import { State } from 'ts/redux/reducer';
 import { colors } from 'ts/style/colors';
+import { AccountReady } from 'ts/types';
 import { constants } from 'ts/utils/constants';
+import { errorReporter } from 'ts/utils/error_reporter';
 
-interface Props {
+interface ModalVoteProps {
     theme?: GlobalStyle;
     isOpen?: boolean;
     onDismiss?: () => void;
-    onWalletConnected?: (providerName: string) => void;
     onVoted?: (voteInfo: VoteInfo) => void;
     zeipId: number;
 }
 
-interface State {
-    currentBalance: BigNumber;
-    isWalletConnected: boolean;
-    providerName: string;
-    isSubmitting: boolean;
-    isSuccessful: boolean;
-    isErrorModalOpen?: boolean;
-    isVoted: boolean;
-    votePreference: string | null;
-    voteHash?: string;
-    signedVote?: any;
-    errorMessage?: string;
-    errors: ErrorProps;
-    web3Wrapper?: Web3Wrapper;
-    contractWrappers?: ContractWrappers;
-    providerEngine?: ZeroExProvider;
-    web3?: any;
-    selectedAddress?: string;
-}
-
 interface FormProps {
     isSuccessful?: boolean;
     isSubmitting?: boolean;
 }
 
-interface ErrorProps {
-    [key: string]: string;
-}
-
-// This is a copy of the generic form and includes a number of extra fields
-// TODO remove the extraneous fields
-export class ModalVote extends React.Component<Props> {
-    public networkId: number;
-    public state: State = {
-        currentBalance: new BigNumber(0),
-        isWalletConnected: false,
-        providerName: 'Metamask',
-        selectedAddress: null,
-        isSubmitting: false,
-        providerEngine: null,
-        isSuccessful: false,
-        isVoted: false,
-        votePreference: null,
-        errors: {},
-    };
-    // shared fields
-    public constructor(props: Props) {
-        super(props);
-    }
-    public render(): React.ReactNode {
-        const { isOpen, onDismiss, zeipId } = this.props;
-        const { isSuccessful, selectedAddress, currentBalance, isErrorModalOpen, errorMessage } = this.state;
-        const bigNumberFormat = {
-            decimalSeparator: '.',
-            groupSeparator: ',',
-            groupSize: 3,
-            secondaryGroupSize: 0,
-            fractionGroupSeparator: ' ',
-            fractionGroupSize: 0,
+export const ModalVote: React.FC<ModalVoteProps> = ({ zeipId, isOpen, onDismiss, onVoted: onVoteInfoReceived }) => {
+    const providerState = useSelector((state: State) => state.providerState);
+    const networkId = useSelector((state: State) => state.networkId);
+    const account = providerState.account as AccountReady;
+    const dispatch = useDispatch();
+    const apiClient = useAPIClient(networkId);
+
+    const [currentBalance, setCurrentBalance] = React.useState(new BigNumber(0));
```

Comment:
NIT: It might be clearer to name this `currentVotingPower` as this is different from the user's ZRX balance in the case of pool operators.
```suggestion
    const [currentVotingPower, setCurrentVotingPower] = React.useState(new BigNumber(0));
```